colossalai/shardformer/inference/engine.py

```diff
@@ -0,0 +1,245 @@
+from functools import partial
+from types import MethodType
+from typing import Any, Callable, List, Optional, Set, Union
+
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+from transformers.generation import GenerationConfig
+from transformers.generation.stopping_criteria import StoppingCriteria, StoppingCriteriaList
+from transformers.tokenization_utils_base import BatchEncoding
+
+from colossalai.cluster import ProcessGroupMesh
+from colossalai.shardformer import ShardConfig, ShardFormer
+from colossalai.shardformer.inference import BatchInferState, MemoryManager
+# from colossalai.shardformer.policies.bloom import BloomModelInferPolicy
+from colossalai.shardformer.policies.auto_policy import get_autopolicy
+
+DP_AXIS, PP_AXIS, TP_AXIS = 0, 1, 2
+
+
+class InferenceEngine:
```

Comment:
change name into TpInferEngine :

Reply:
Gotcha. Will do while revising a few bugs of bloom modelling.

Reply:
Updated